sdk/core/azure-core-http-netty/src/test/java/com/azure/core/http/netty/ReactorNettyClientTests.java

```diff
@@ -281,6 +291,102 @@ public void testConcurrentRequests() {
         Assertions.fail("Method needs to be reimplemented");
     }
 
+    /**
+     * Tests that deep copying the buffers returned by Netty will make the stream returned to the customer resilient to
+     * Netty reclaiming them once the 'onNext' operator chain has completed.
+     */
+    @Test
+    public void deepCopyBufferConfiguredInBuilder() {
+        HttpClient client = new NettyAsyncHttpClientBuilder().disableBufferCopy(false).build();
+
+        HttpResponse response = client.send(new HttpRequest(HttpMethod.GET, url(server, LONG_BODY_PATH))).block();
+        assertNotNull(response);
+        assertEquals(200, response.getStatusCode());
+
+        DelayWriteStream delayWriteStream = new DelayWriteStream();
+        response.getBody().doOnNext(delayWriteStream::write).blockLast();
+        assertEquals(LONG_BODY, delayWriteStream.aggregateAsString());
+    }
+
+    /**
+     * Tests that preventing deep copying the buffers returned by Netty won't make the stream returned to the customer
+     * resilient to Netty reclaiming them once the 'onNext' operator chain has completed.
+     */
+    @Test
+    public void ignoreDeepCopyBufferConfiguredInBuilder() {
```

Comment:
Since these two methods `deepCopyBufferConfiguredInBuilder` and `ignoreDeepCopyBufferConfiguredInBuilder`  only differs in boolean values for `disableBufferCopy`. Could we use dataprovider to pass the Boolean and just have one method which test both use cases ?